### PR TITLE
AI should avoid splitting troops on easy difficulty

### PR DESCRIPTION
**Pull request by idshibanov**
Might be up for discussion but removing the optimization should make battles easier. It will be a noticeable difference in behaviour between difficulties.

### CODE REVIEW

**Review by oleg-derevenetz — APPROVED** [review]
Technically this change is OK. However, I do not know how much this will make the task easier for inexperienced players on easy difficulties :)

File: src/fheroes2/game/difficulty.cpp [hunk 1]
```diff
@@ -158,3 +158,8 @@ uint32_t Difficulty::GetDimensionDoorLimit( int difficulty )
     }
     return UINT32_MAX;
 }
+
+bool Difficulty::AISplittingWeakStacks( const int difficulty )
+{
+    return difficulty > Difficulty::NORMAL;
```
Comment (ihhub) [inline src/fheroes2/game/difficulty.cpp:164]: I recommend to still use switch case so it is easier to read the code and it handles all possible difficulties (if any new in the future as well).

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @idshibanov , I left 2 minor comments here. Could you please take a look?

Also I don't believe that it makes any significant difference in the gameplay but let's hear others.

**Review by Branikolog — APPROVED** [review]

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@idshibanov , huge thanks for this change!

**Pull request merged**